library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,195 @@
+#include "L1_Peripheral/lpc40xx/system_controller.hpp"
+#include "L1_Peripheral/lpc40xx/timer.hpp"
+#include "L1_Peripheral/lpc40xx/capture.hpp"
+#include "L1_Peripheral/timer.hpp"
+#include "L1_Peripheral/capture.hpp"
+#include "utility/log.hpp"
+#include "utility/status.hpp"
+#include "utility/units.hpp"
+
+
+namespace sjsu
+{
+namespace lpc40xx
+{
+class Capture final : public sjsu::Capture
+{
+ public:
+    template<const Timer::ChannelPartial_t &port>
+    static void TimerHandler()
+    {
+        TimerHandler(port);
+    }
+
+    struct Channel // NOLINT
+    {
+     private:
+        inline static IsrPointer timer0_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial0 = {
+        .timer_register = LPC_TIM0,
+        .power_id       = SystemController::Peripherals::kTimer0,
+        .irq            = IRQn::TIMER0_IRQn,
+        .user_callback  = &timer0_isr,
+        };
+
+        inline static IsrPointer timer1_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial1 = {
+        .timer_register = LPC_TIM1,
+        .power_id       = SystemController::Peripherals::kTimer1,
+        .irq            = IRQn::TIMER1_IRQn,
+        .user_callback  = &timer1_isr,
+        };
+
+        inline static IsrPointer timer2_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial2 = {
+        .timer_register = LPC_TIM2,
+        .power_id       = SystemController::Peripherals::kTimer2,
+        .irq            = IRQn::TIMER2_IRQn,
+        .user_callback  = &timer2_isr,
+        };
+
+        inline static IsrPointer timer3_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial3 = {
+        .timer_register = LPC_TIM3,
+        .power_id       = SystemController::Peripherals::kTimer3,
+        .irq            = IRQn::TIMER3_IRQn,
+        .user_callback  = &timer3_isr,
+        };
+
+     public:
+        inline static const Timer::Channel_t kCaptureTimer0 =
+        {
+            .channel = kTimerPartial0,
+            .isr = TimerHandler<kTimerPartial0>
+        };
+        inline static const Timer::Channel_t kCaptureTimer1 =
+        {
+            .channel = kTimerPartial1,
+            .isr = TimerHandler<kTimerPartial1>
+        };
+        inline static const Timer::Channel_t kCaptureTimer2 =
+        {
+            .channel = kTimerPartial2,
+            .isr = TimerHandler<kTimerPartial2>
+        };
+        inline static const Timer::Channel_t kCaptureTimer3 =
+        {
+            .channel = kTimerPartial3,
+            .isr = TimerHandler<kTimerPartial3>
+        };
+    };  // struct Channel
+
+
+    static void TimerHandler(const Timer::ChannelPartial_t & channel)
+    {
+        if (*channel.user_callback != nullptr)
+        {
+            (*channel.user_callback)();
+        }
+        channel.timer_register->IR |= 0x30;
+    }
+
+    typedef struct
```

Comment:
Do not use `typedef` in C++. Prefer using Using. Also, there is no need to typedef structs in C++ at all. You can simply just use `struct` like you would `class`.